libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -168,12 +195,33 @@ func (p *pod) GenerateHints(event bus.Event) bus.Event {
 
 // Start starts the eventer
 func (p *pod) Start() error {
+	if p.nodeWatcher != nil {
+		err := p.nodeWatcher.Start()
+		if err != nil {
+			return err
+		}
+	}
+
+	if p.namespaceWatcher != nil {
+		if err := p.namespaceWatcher.Start(); err != nil {
+			return err
+		}
+	}
+
 	return p.watcher.Start()
```

Comment:
since we explicitly ask for enrichment of node and pod, i would assume there should be start up failures if there are any in the enrichment as well. Best effort wouldnt work in this case.